utilities/cli/holohub.py

```diff
@@ -364,14 +401,44 @@ def handle_run_container(self, args: argparse.Namespace) -> None:
         )
 
         container.dryrun = args.dryrun
-        container.build()
+
+        # Get mode-specific build and run arguments if project is specified
+        mode_build_args = ""
+        mode_docker_opts = ""
+        if args.project:
+            project_data = self._find_project(
+                project_name=args.project,
+                language=args.language if hasattr(args, "language") else None,
+            )
+            mode_config = self._get_mode_config(project_data, args.mode)
+
+            # Get mode-specific build args
+            build_config = mode_config.get("build", {})
+            build_docker_args = build_config.get("docker_args", [])
+            if isinstance(build_docker_args, list):
+                mode_build_args = " ".join(build_docker_args)
+
+            # Get mode-specific run args
+            run_config = mode_config.get("run", {})
+            run_docker_args = run_config.get("docker_args", [])
+            if isinstance(run_docker_args, list):
+                mode_docker_opts = " ".join(run_docker_args)
+
+        # Build container with mode-specific args
+        container.build(build_args=mode_build_args if mode_build_args else None)
+
+        # Combine user-provided docker opts with mode-specific opts
+        combined_docker_opts = args.docker_opts
+        if mode_docker_opts:
+            combined_docker_opts = f"{args.docker_opts} {mode_docker_opts}".strip()
```

Comment:
Can we standardize with "docker opts" or "docker args" naming? Seems like they refer to the same thing?